src/Datasource/EntityTrait.php

```diff
@@ -281,8 +299,10 @@ public function &get($property)
 
         if ($this->_methodExists($method)) {
             $result = $this->{$method}($value);
-            return $result;
+            return $recursor($result, $nested);
         }
+
+        $value = $recursor($value, $nested);
```

Comment:
Is there a way we can avoid the additional function calls when the property access is a simple non-dotted path? While it might not seem like much all these extra function calls add up in big apps. I remember that part of the big speedups from 1.3 -> 2.0 was due to calling a few 1000 functions fewer per request.